---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# supersigs

<!-- badges: start -->
[![Lifecycle: experimental](https://img.shields.io/badge/lifecycle-experimental-blue.svg)](https://www.tidyverse.org/lifecycle/#experimental)
<!-- [![CRAN Status](https://www.r-pkg.org/badges/version/pkgdown)](https://cran.r-project.org/package=pkgdown) -->
<!-- [![R build status](https://github.com/r-lib/pkgdown/workflows/R-CMD-check/badge.svg)](https://github.com/r-lib/supersigs/actions) -->
<!-- [![Codecov test coverage](https://codecov.io/gh/r-lib/pkgdown/branch/master/graph/badge.svg)](https://codecov.io/gh/r-lib/supersigs?branch=master) -->
<!-- badges: end -->

`supersigs` is a companion R package to a method proposed by *Afsari, et al. (2021, ELife)* to generate mutational signatures from single nucleotide variants in the cancer genome. **Note: Package is under active development.**

More details on the statistical method can be found in this paper:

* Afsari, B., Kuo, A., Zhang, Y., Li, L., Lahouel, K., Danilova, L., Favorov, A., Rosenquist, T. A., Grollman, A. P., Kinzler, K. W., Cope, L., Vogelstein, B., & Tomasetti, C. (2021). Supervised mutational signatures for obesity and other tissue-specific etiological factors in cancer. ELife, 10. [https://doi.org/10.7554/elife.61082](https://doi.org/10.7554/eLife.61082)

## Installation

You can install the development version of supersigs from github using the `install_github()` function from the `devtools` package.

```{r eval = F}
# Install development version from GitHub
devtools::install_github("TomasettiLab/supersigs")
```

## Data Format

At a minimum, the data you will need are the age and mutations for each patient. An example is provided below.
(Note that you will need to process the data before running the core functions, see `vignette("supersigs")`
for details.)

```{r echo = F}
library(supersigs)
head(example_dt)
```

## Core functions

In brief, the `supersigs` package contains three core functions: `get_signature`, `predict_signature`, and `partial_signature`.

`get_signature` trains a supervised signature for a given factor (e.g. smoking).

```{r eval = F}
supersig <- get_signature(data = data, factor = "smoking", wgs = F)
```

`predict_signature` uses the trained supervised signature to obtain predicted probabilities (e.g. probability of smoker) on a new dataset.

```{r eval = F}
pred <- predict_signature(object = supersig, newdata = data, factor = "smoking")
```

`partial_signature` removes the contribution of a trained signature from the dataset.

```{r eval = F}
data <- partial_signature(data = data, object = supersig)
```

```{r include = F, eval = F}
devtools::build_readme()
```

## Tutorial

To follow a tutorial on how to use the package, see `vignette("supersigs")` (or type `vignette("supersigs")` in R).
