<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using supersigs • supersigs</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using supersigs">
<meta property="og:description" content="supersigs">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-HRWJPS4FR0"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-HRWJPS4FR0');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">supersigs</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/supersigs.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/TomasettiLab/supersigs">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="supersigs_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using supersigs</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/TomasettiLab/supersigs/blob/master/vignettes/supersigs.Rmd"><code>vignettes/supersigs.Rmd</code></a></small>
      <div class="hidden name"><code>supersigs.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tomasettilab.github.io/supersigs/">supersigs</a></span><span class="op">)</span></code></pre></div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The <code>supersigs</code> package implements the supervised method proposed by <em>Afsari, et al.</em> to find signatures (“SuperSigs”). In this vignette, we cover how to preprocess your data and run the method in <code>supersigs</code>.</p>
</div>
<div id="preprocessing-your-data" class="section level1">
<h1 class="hasAnchor">
<a href="#preprocessing-your-data" class="anchor"></a>Preprocessing your data</h1>
<div id="vcf-file" class="section level2">
<h2 class="hasAnchor">
<a href="#vcf-file" class="anchor"></a>VCF file</h2>
<p>If you have a VCF file, you can use <code>readVcf</code> from the <code>VariantAnnotation</code> package to read in your VCF file as a VCF object. The age of each patient should be stored as <code>age</code> in the <code>colData</code> of your <code>VCF</code> object. Then use <code>process_vcf</code> to transform the VCF object into a simplified data frame format, which will be explained further in <a href="#example-data">Example data</a>.</p>
<p>If you do not have a VCF file, skip to <a href="#example-data">Example data</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load packages for make_matrix function</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">VariantAnnotation</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="va">fl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"chr22.vcf.gz"</span>, package<span class="op">=</span><span class="st">"VariantAnnotation"</span><span class="op">)</span>
<span class="va">vcf</span> <span class="op">&lt;-</span> <span class="fu">VariantAnnotation</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/readVcf-methods.html">readVcf</a></span><span class="op">(</span><span class="va">fl</span>, <span class="st">"hg19"</span><span class="op">)</span> 
<span class="co"># Subset to first sample</span>
<span class="va">vcf</span> <span class="op">&lt;-</span> <span class="va">vcf</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>
<span class="co"># Subset to row positions with homozygous or heterozygous alt</span>
<span class="va">positions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html">geno</a></span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span><span class="op">$</span><span class="va">GT</span> <span class="op">!=</span> <span class="st">"0|0"</span> 
<span class="va">vcf</span> <span class="op">&lt;-</span> <span class="va">vcf</span><span class="op">[</span><span class="va">positions</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>,<span class="op">]</span>
<span class="fu">colData</span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span><span class="op">$</span><span class="va">age</span> <span class="op">&lt;-</span> <span class="fl">50</span>    <span class="co"># Add patient age to colData</span>
<span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/process_vcf.html">process_vcf</a></span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dt</span><span class="op">)</span>
<span class="co">#&gt;   sample_id age chromosome position ref alt</span>
<span class="co">#&gt; 1   HG00096  50      chr22 50326116   C   T</span>
<span class="co">#&gt; 2   HG00096  50      chr22 50336761   G   A</span>
<span class="co">#&gt; 3   HG00096  50      chr22 50346072   C   T</span>
<span class="co">#&gt; 4   HG00096  50      chr22 50350418   T   C</span>
<span class="co">#&gt; 5   HG00096  50      chr22 50351413   C   T</span>
<span class="co">#&gt; 6   HG00096  50      chr22 50351977   G   A</span></code></pre></div>
</div>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h2>
<p>The method uses single-base mutations in exomic data from cancer samples. Specifically, it requires data on every sample’s mutations, the positions of those mutations, and the age of all patients. This data can be represented as a list of mutations. Below is an example dataset (stored and accessible from the <code>supersigs</code> R package). If you have a VCF file, read the <a href="#vcf-file">VCF file</a> section to see how to process your data into the following format.</p>
<ul>
<li>
<code>sample_id</code> is an ID for each sample</li>
<li>
<code>age</code> is the age of the patient</li>
<li>
<code>chromosome</code> and <code>position</code> is the position of the mutation</li>
<li>
<code>ref</code> is the original nucleotide</li>
<li>
<code>alt</code> is the mutated nucleotide</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">example_dt</span><span class="op">)</span>
<span class="co">#&gt;   sample_id age chromosome  position ref alt</span>
<span class="co">#&gt; 1         1  50       chr1  94447621   G   C</span>
<span class="co">#&gt; 2         1  50       chr2 202005395   A   C</span>
<span class="co">#&gt; 3         1  50       chr7  20784978   T   A</span>
<span class="co">#&gt; 4         1  50       chr7  87179255   C   G</span>
<span class="co">#&gt; 5         1  50      chr19   1059712   G   T</span>
<span class="co">#&gt; 6         2  55       chr1  76226977   T   C</span></code></pre></div>
</div>
<div id="transform-data" class="section level2">
<h2 class="hasAnchor">
<a href="#transform-data" class="anchor"></a>Transform data</h2>
<p>Once you’ve read in your data, you will need to transform it into a data frame of features before running the core functions. This involves 2 steps:</p>
<ol style="list-style-type: decimal">
<li><p>First, we assume that mutations are the same regardless of the strand on which it occurred. For example, this means that C&gt;A mutations are considered the same as G&gt;T mutations and we will convert all G&gt;T mutations to be denoted as C&gt;A mutations.</p></li>
<li><p>Because the features used are built upon trinucleotide features (e.g. A[C&gt;A]T), this will require matching your mutations to a reference genome to identify what the flanking bases of every mutation are. In our example below, we will use the <code>hg19</code> reference genome.</p></li>
</ol>
<p>Both of these steps are done by the <code>make_matrix</code> function. Note that using the <code>make_matrix</code> function requires installing and loading a reference genome (<code>BSgenome.Hsapiens.UCSC.hg19</code> and <code>BSgenome.Hsapiens.UCSC.hg38</code> are supported).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load packages for make_matrix function</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg19</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<p>We apply <code>make_matrix</code> to transform our example dataset (<code>example_dt</code>) into a data frame of trinucleotide mutations (<code>input_dt</code>), which is the format required by the <code>supersigs</code> R package. Each row in <code>input_dt</code> corresponds to a different patient and the values in the columns are the number of mutations for each trinucleotide mutation.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">input_dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_matrix.html">make_matrix</a></span><span class="op">(</span><span class="va">example_dt</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">input_dt</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 5 x 98</span>
<span class="co">#&gt;   sample_id   age `A[T&gt;G]T` `C[T&gt;A]A` `G[C&gt;A]A` `G[C&gt;G]G` `G[C&gt;G]T` `A[C&gt;G]T`</span>
<span class="co">#&gt;       &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt; 1         1    50         1         1         1         1         1         0</span>
<span class="co">#&gt; 2         2    55         0         0         0         0         0         1</span>
<span class="co">#&gt; 3         3    72         1         1         0         0         0         0</span>
<span class="co">#&gt; 4         4    53         0         0         0         0         0         1</span>
<span class="co">#&gt; 5         5    48         0         1         1         0         0         0</span>
<span class="co">#&gt; # … with 90 more variables: C[C&gt;A]T &lt;dbl&gt;, C[T&gt;C]G &lt;dbl&gt;, T[C&gt;A]C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[C&gt;A]T &lt;dbl&gt;, A[T&gt;C]C &lt;dbl&gt;, C[T&gt;C]C &lt;dbl&gt;, T[T&gt;A]C &lt;dbl&gt;, C[C&gt;G]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   G[C&gt;T]A &lt;dbl&gt;, A[C&gt;A]T &lt;dbl&gt;, C[C&gt;A]C &lt;dbl&gt;, G[T&gt;G]T &lt;dbl&gt;, C[C&gt;T]C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[C&gt;T]C &lt;dbl&gt;, A[C&gt;T]C &lt;dbl&gt;, G[C&gt;T]C &lt;dbl&gt;, C[C&gt;T]T &lt;dbl&gt;, T[C&gt;T]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[C&gt;T]T &lt;dbl&gt;, G[C&gt;T]T &lt;dbl&gt;, C[C&gt;T]A &lt;dbl&gt;, T[C&gt;T]A &lt;dbl&gt;, A[C&gt;T]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[C&gt;T]G &lt;dbl&gt;, T[C&gt;T]G &lt;dbl&gt;, A[C&gt;T]G &lt;dbl&gt;, G[C&gt;T]G &lt;dbl&gt;, A[C&gt;A]C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   G[C&gt;A]C &lt;dbl&gt;, G[C&gt;A]T &lt;dbl&gt;, C[C&gt;A]A &lt;dbl&gt;, T[C&gt;A]A &lt;dbl&gt;, A[C&gt;A]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[C&gt;A]G &lt;dbl&gt;, T[C&gt;A]G &lt;dbl&gt;, A[C&gt;A]G &lt;dbl&gt;, G[C&gt;A]G &lt;dbl&gt;, C[C&gt;G]C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[C&gt;G]C &lt;dbl&gt;, A[C&gt;G]C &lt;dbl&gt;, G[C&gt;G]C &lt;dbl&gt;, C[C&gt;G]T &lt;dbl&gt;, T[C&gt;G]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[C&gt;G]A &lt;dbl&gt;, T[C&gt;G]A &lt;dbl&gt;, A[C&gt;G]A &lt;dbl&gt;, G[C&gt;G]A &lt;dbl&gt;, T[C&gt;G]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[C&gt;G]G &lt;dbl&gt;, T[T&gt;C]C &lt;dbl&gt;, G[T&gt;C]C &lt;dbl&gt;, C[T&gt;C]T &lt;dbl&gt;, T[T&gt;C]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[T&gt;C]T &lt;dbl&gt;, G[T&gt;C]T &lt;dbl&gt;, C[T&gt;C]A &lt;dbl&gt;, T[T&gt;C]A &lt;dbl&gt;, A[T&gt;C]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   G[T&gt;C]A &lt;dbl&gt;, T[T&gt;C]G &lt;dbl&gt;, A[T&gt;C]G &lt;dbl&gt;, G[T&gt;C]G &lt;dbl&gt;, C[T&gt;A]C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[T&gt;A]C &lt;dbl&gt;, G[T&gt;A]C &lt;dbl&gt;, C[T&gt;A]T &lt;dbl&gt;, T[T&gt;A]T &lt;dbl&gt;, A[T&gt;A]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   G[T&gt;A]T &lt;dbl&gt;, T[T&gt;A]A &lt;dbl&gt;, A[T&gt;A]A &lt;dbl&gt;, G[T&gt;A]A &lt;dbl&gt;, C[T&gt;A]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[T&gt;A]G &lt;dbl&gt;, A[T&gt;A]G &lt;dbl&gt;, G[T&gt;A]G &lt;dbl&gt;, C[T&gt;G]C &lt;dbl&gt;, T[T&gt;G]C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[T&gt;G]C &lt;dbl&gt;, G[T&gt;G]C &lt;dbl&gt;, C[T&gt;G]T &lt;dbl&gt;, T[T&gt;G]T &lt;dbl&gt;, C[T&gt;G]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[T&gt;G]A &lt;dbl&gt;, A[T&gt;G]A &lt;dbl&gt;, G[T&gt;G]A &lt;dbl&gt;, C[T&gt;G]G &lt;dbl&gt;, T[T&gt;G]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[T&gt;G]G &lt;dbl&gt;, G[T&gt;G]G &lt;dbl&gt;</span></code></pre></div>
</div>
</div>
<div id="getting-your-signature" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-your-signature" class="anchor"></a>Getting your signature</h1>
<p>To apply the supervised method on your data, run the <code>get_signature</code> function. The function has two parameters: an input data frame <code>data</code> and the <code>factor</code> (e.g. <code>factor = "Smoking"</code>). <code>data</code> is a data frame with the following columns:</p>
<ul>
<li>
<code>IndVar</code> (indicator variable) is a logical indicator for whether they were exposed to the <code>factor</code> or not</li>
<li>
<code>sample_id</code> is an ID for each sample</li>
<li>
<code>age</code> is the age of the patient</li>
<li>columns of counts for all 96 trinucleotide mutations</li>
</ul>
<p>The process of converting a VCF file to this format is covered in <a href="#preprocessing-your-data">Preprocessing your data</a>. An example for <code>data</code> is printed below.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># Add IndVar column</span>
<span class="va">input_dt</span> <span class="op">&lt;-</span> <span class="va">input_dt</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>IndVar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html">relocate</a></span><span class="op">(</span><span class="va">IndVar</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">input_dt</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 5 x 99</span>
<span class="co">#&gt;   IndVar sample_id   age `A[T&gt;G]T` `C[T&gt;A]A` `G[C&gt;A]A` `G[C&gt;G]G` `G[C&gt;G]T`</span>
<span class="co">#&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt; 1      1         1    50         1         1         1         1         1</span>
<span class="co">#&gt; 2      1         2    55         0         0         0         0         0</span>
<span class="co">#&gt; 3      1         3    72         1         1         0         0         0</span>
<span class="co">#&gt; 4      0         4    53         0         0         0         0         0</span>
<span class="co">#&gt; 5      0         5    48         0         1         1         0         0</span>
<span class="co">#&gt; # … with 91 more variables: A[C&gt;G]T &lt;dbl&gt;, C[C&gt;A]T &lt;dbl&gt;, C[T&gt;C]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[C&gt;A]C &lt;dbl&gt;, T[C&gt;A]T &lt;dbl&gt;, A[T&gt;C]C &lt;dbl&gt;, C[T&gt;C]C &lt;dbl&gt;, T[T&gt;A]C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[C&gt;G]G &lt;dbl&gt;, G[C&gt;T]A &lt;dbl&gt;, A[C&gt;A]T &lt;dbl&gt;, C[C&gt;A]C &lt;dbl&gt;, G[T&gt;G]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[C&gt;T]C &lt;dbl&gt;, T[C&gt;T]C &lt;dbl&gt;, A[C&gt;T]C &lt;dbl&gt;, G[C&gt;T]C &lt;dbl&gt;, C[C&gt;T]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[C&gt;T]T &lt;dbl&gt;, A[C&gt;T]T &lt;dbl&gt;, G[C&gt;T]T &lt;dbl&gt;, C[C&gt;T]A &lt;dbl&gt;, T[C&gt;T]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[C&gt;T]A &lt;dbl&gt;, C[C&gt;T]G &lt;dbl&gt;, T[C&gt;T]G &lt;dbl&gt;, A[C&gt;T]G &lt;dbl&gt;, G[C&gt;T]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[C&gt;A]C &lt;dbl&gt;, G[C&gt;A]C &lt;dbl&gt;, G[C&gt;A]T &lt;dbl&gt;, C[C&gt;A]A &lt;dbl&gt;, T[C&gt;A]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[C&gt;A]A &lt;dbl&gt;, C[C&gt;A]G &lt;dbl&gt;, T[C&gt;A]G &lt;dbl&gt;, A[C&gt;A]G &lt;dbl&gt;, G[C&gt;A]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[C&gt;G]C &lt;dbl&gt;, T[C&gt;G]C &lt;dbl&gt;, A[C&gt;G]C &lt;dbl&gt;, G[C&gt;G]C &lt;dbl&gt;, C[C&gt;G]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[C&gt;G]T &lt;dbl&gt;, C[C&gt;G]A &lt;dbl&gt;, T[C&gt;G]A &lt;dbl&gt;, A[C&gt;G]A &lt;dbl&gt;, G[C&gt;G]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[C&gt;G]G &lt;dbl&gt;, A[C&gt;G]G &lt;dbl&gt;, T[T&gt;C]C &lt;dbl&gt;, G[T&gt;C]C &lt;dbl&gt;, C[T&gt;C]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[T&gt;C]T &lt;dbl&gt;, A[T&gt;C]T &lt;dbl&gt;, G[T&gt;C]T &lt;dbl&gt;, C[T&gt;C]A &lt;dbl&gt;, T[T&gt;C]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[T&gt;C]A &lt;dbl&gt;, G[T&gt;C]A &lt;dbl&gt;, T[T&gt;C]G &lt;dbl&gt;, A[T&gt;C]G &lt;dbl&gt;, G[T&gt;C]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[T&gt;A]C &lt;dbl&gt;, A[T&gt;A]C &lt;dbl&gt;, G[T&gt;A]C &lt;dbl&gt;, C[T&gt;A]T &lt;dbl&gt;, T[T&gt;A]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[T&gt;A]T &lt;dbl&gt;, G[T&gt;A]T &lt;dbl&gt;, T[T&gt;A]A &lt;dbl&gt;, A[T&gt;A]A &lt;dbl&gt;, G[T&gt;A]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[T&gt;A]G &lt;dbl&gt;, T[T&gt;A]G &lt;dbl&gt;, A[T&gt;A]G &lt;dbl&gt;, G[T&gt;A]G &lt;dbl&gt;, C[T&gt;G]C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[T&gt;G]C &lt;dbl&gt;, A[T&gt;G]C &lt;dbl&gt;, G[T&gt;G]C &lt;dbl&gt;, C[T&gt;G]T &lt;dbl&gt;, T[T&gt;G]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[T&gt;G]A &lt;dbl&gt;, T[T&gt;G]A &lt;dbl&gt;, A[T&gt;G]A &lt;dbl&gt;, G[T&gt;G]A &lt;dbl&gt;, C[T&gt;G]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[T&gt;G]G &lt;dbl&gt;, A[T&gt;G]G &lt;dbl&gt;, G[T&gt;G]G &lt;dbl&gt;</span></code></pre></div>
<p>Once you have the correct data format, apply <code>get_signature</code> to the dataset to get your <code>SuperSig</code>, which is an S4 object containing four slots:</p>
<ul>
<li>
<code>Signature</code> is the signature, represented as their differences in mean rates (or the overall mean rate if the factor is “age”) between the two groups (exposed versus unexposed)</li>
<li>
<code>Features</code> is the list of features that comprise the signature and their representation in terms of the fundamental (trinucleotide) mutations</li>
<li>
<code>AUC</code> is the apparent AUC of the model (i.e. not cross-validated)</li>
<li>
<code>Model</code> is the list containing the trained logistic regression model (glm class)</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">supersig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_signature.html">get_signature</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">input_dt</span>, factor <span class="op">=</span> <span class="st">"Smoking"</span><span class="op">)</span>
<span class="co">#&gt; Begin feature engineering...</span>
<span class="co">#&gt; Begin cross-validated selection over 4 features and 15 inner folds...</span>
<span class="co">#&gt; ...testing inner fold 1</span>
<span class="co">#&gt; ...testing inner fold 2</span>
<span class="co">#&gt; ...testing inner fold 3</span>
<span class="co">#&gt; ...testing inner fold 4</span>
<span class="co">#&gt; ...testing inner fold 5</span>
<span class="co">#&gt; ...testing inner fold 6</span>
<span class="co">#&gt; ...testing inner fold 7</span>
<span class="co">#&gt; ...testing inner fold 8</span>
<span class="co">#&gt; ...testing inner fold 9</span>
<span class="co">#&gt; ...testing inner fold 10</span>
<span class="co">#&gt; ...testing inner fold 11</span>
<span class="co">#&gt; ...testing inner fold 12</span>
<span class="co">#&gt; ...testing inner fold 13</span>
<span class="co">#&gt; ...testing inner fold 14</span>
<span class="co">#&gt; ...testing inner fold 15</span>
<span class="va">supersig</span>
<span class="co">#&gt; Signature:</span>
<span class="co">#&gt;              X1</span>
<span class="co">#&gt; 1 -0.0007475199</span>
<span class="co">#&gt; Features:</span>
<span class="co">#&gt; $X1</span>
<span class="co">#&gt;       F11       F12       F13       F14      F117      F118      F119      F120 </span>
<span class="co">#&gt; "A[C&gt;A]A" "A[C&gt;A]C" "A[C&gt;A]G" "A[C&gt;A]T" "A[C&gt;G]A" "A[C&gt;G]C" "A[C&gt;G]G" "A[C&gt;G]T" </span>
<span class="co">#&gt;      F132      F133      F134      F135      F148      F149      F150      F151 </span>
<span class="co">#&gt; "A[C&gt;T]A" "A[C&gt;T]C" "A[C&gt;T]G" "A[C&gt;T]T" "A[T&gt;A]A" "A[T&gt;A]C" "A[T&gt;A]G" "A[T&gt;A]T" </span>
<span class="co">#&gt;      F163      F164      F165      F166      F179      F180      F181       F15 </span>
<span class="co">#&gt; "A[T&gt;C]A" "A[T&gt;C]C" "A[T&gt;C]G" "A[T&gt;C]T" "A[T&gt;G]A" "A[T&gt;G]C" "A[T&gt;G]G" "C[C&gt;A]A" </span>
<span class="co">#&gt;       F16       F17       F18      F121      F122      F123      F124      F136 </span>
<span class="co">#&gt; "C[C&gt;A]C" "C[C&gt;A]G" "C[C&gt;A]T" "C[C&gt;G]A" "C[C&gt;G]C" "C[C&gt;G]G" "C[C&gt;G]T" "C[C&gt;T]A" </span>
<span class="co">#&gt;      F137      F138      F139      F152      F153      F154      F167      F168 </span>
<span class="co">#&gt; "C[C&gt;T]C" "C[C&gt;T]G" "C[C&gt;T]T" "C[T&gt;A]C" "C[T&gt;A]G" "C[T&gt;A]T" "C[T&gt;C]A" "C[T&gt;C]C" </span>
<span class="co">#&gt;      F169      F170      F182      F183      F184      F185       F19      F110 </span>
<span class="co">#&gt; "C[T&gt;C]G" "C[T&gt;C]T" "C[T&gt;G]A" "C[T&gt;G]C" "C[T&gt;G]G" "C[T&gt;G]T" "G[C&gt;A]A" "G[C&gt;A]C" </span>
<span class="co">#&gt;      F111      F112      F125      F126      F127      F140      F141      F142 </span>
<span class="co">#&gt; "G[C&gt;A]G" "G[C&gt;A]T" "G[C&gt;G]A" "G[C&gt;G]C" "G[C&gt;G]T" "G[C&gt;T]A" "G[C&gt;T]C" "G[C&gt;T]G" </span>
<span class="co">#&gt;      F143      F155      F156      F157      F158      F171      F172      F173 </span>
<span class="co">#&gt; "G[C&gt;T]T" "G[T&gt;A]A" "G[T&gt;A]C" "G[T&gt;A]G" "G[T&gt;A]T" "G[T&gt;C]A" "G[T&gt;C]C" "G[T&gt;C]G" </span>
<span class="co">#&gt;      F174      F186      F187      F188      F189      F113      F114      F115 </span>
<span class="co">#&gt; "G[T&gt;C]T" "G[T&gt;G]A" "G[T&gt;G]C" "G[T&gt;G]G" "G[T&gt;G]T" "T[C&gt;A]A" "T[C&gt;A]C" "T[C&gt;A]G" </span>
<span class="co">#&gt;      F116      F128      F129      F130      F131      F144      F145      F146 </span>
<span class="co">#&gt; "T[C&gt;A]T" "T[C&gt;G]A" "T[C&gt;G]C" "T[C&gt;G]G" "T[C&gt;G]T" "T[C&gt;T]A" "T[C&gt;T]C" "T[C&gt;T]G" </span>
<span class="co">#&gt;      F147      F159      F160      F161      F162      F175      F176      F177 </span>
<span class="co">#&gt; "T[C&gt;T]T" "T[T&gt;A]A" "T[T&gt;A]C" "T[T&gt;A]G" "T[T&gt;A]T" "T[T&gt;C]A" "T[T&gt;C]C" "T[T&gt;C]G" </span>
<span class="co">#&gt;      F178      F190      F191      F192      F193 </span>
<span class="co">#&gt; "T[T&gt;C]T" "T[T&gt;G]A" "T[T&gt;G]C" "T[T&gt;G]G" "T[T&gt;G]T" </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Model:</span>
<span class="co">#&gt; $Logit</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Call:  glm(formula = IndVar ~ ., family = binomial(), data = x)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Coefficients:</span>
<span class="co">#&gt; (Intercept)           X1  </span>
<span class="co">#&gt;       7.118      -86.601  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Degrees of Freedom: 4 Total (i.e. Null);  3 Residual</span>
<span class="co">#&gt; Null Deviance:       6.73 </span>
<span class="co">#&gt; Residual Deviance: 4.279     AIC: 8.279</span></code></pre></div>
<p>To obtain a signature representation that is more interpretable, you can group the trinucleotide features within each feature using the <code>simplify_signature</code> function (with an option to use <a href="https://www.bioinformatics.org/sms/iupac.html">IUPAC</a> labels). This is useful for making plots of signatures.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">features</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simplify_signature.html">simplify_signature</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">supersig</span>, iupac <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">features_iupac</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simplify_signature.html">simplify_signature</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">supersig</span>, iupac <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">features_iupac</span><span class="op">)</span>,
           differences <span class="op">=</span> <span class="va">features_iupac</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">features</span>, y <span class="op">=</span> <span class="va">differences</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="supersigs_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div id="using-a-signature" class="section level1">
<h1 class="hasAnchor">
<a href="#using-a-signature" class="anchor"></a>Using a signature</h1>
<p>To apply the <code>SuperSig</code> to a new dataset, use the <code>predict_signature</code> function. This function returns the new dataset with columns for feature counts for the signature and a score column for the predicted classification score.</p>
<p>Below is an example for the <code>SuperSig</code> we trained in the previous section. We reuse <code>input_dt</code> as our “new data” for illustrative purposes, but in practice, you would use a different dataset from the one that was used to train the signature (e.g. a test set).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">newdata</span> <span class="op">=</span> <span class="fu"><a href="../reference/predict_signature.html">predict_signature</a></span><span class="op">(</span><span class="va">supersig</span>, newdata <span class="op">=</span> <span class="va">input_dt</span>, factor <span class="op">=</span> <span class="st">"Smoking"</span><span class="op">)</span>

<span class="va">newdata</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">X1</span>, <span class="va">score</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 5 x 2</span>
<span class="co">#&gt;       X1 score</span>
<span class="co">#&gt;    &lt;dbl&gt; &lt;dbl&gt;</span>
<span class="co">#&gt; 1 0.04   0.975</span>
<span class="co">#&gt; 2 0.0909 0.320</span>
<span class="co">#&gt; 3 0.0417 0.971</span>
<span class="co">#&gt; 4 0.0943 0.259</span>
<span class="co">#&gt; 5 0.0833 0.475</span></code></pre></div>
<p>In addition, you may wish to use a <code>SuperSig</code> pre-trained on TCGA data. These are accessible from the package in <code>supersig_ls</code>, where each element of the list is a <code>SuperSig</code>. There are 67 SuperSigs that have been trained on various tissues and factors. The names are printed below (formatted as “factor (tissue)”). Details regarding the training of these signatures are discussed in <em>Afsari, et al. (2021, ELife)</em>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">supersig_ls</span><span class="op">)</span>
<span class="co">#&gt;  [1] "AGE (LAML)"      "AGE (BLCA)"      "AGE (LUAD)"      "AGE (LGG)"      </span>
<span class="co">#&gt;  [5] "AGE (HNSCC)"     "AGE (KIRC)"      "AGE (KIRP)"      "AGE (KICH)"     </span>
<span class="co">#&gt;  [9] "AGE (LIHC)"      "AGE (STAD)"      "AGE (THCA)"      "AGE (UVM)"      </span>
<span class="co">#&gt; [13] "AGE (SKCM)"      "AGE (ACC)"       "AGE (CHOL)"      "AGE (GBM)"      </span>
<span class="co">#&gt; [17] "AGE (CESC)"      "AGE (COAD)"      "AGE (PCPG)"      "AGE (PAAD)"     </span>
<span class="co">#&gt; [21] "AGE (PRAD)"      "AGE (ESCSQ)"     "AGE (ESCAD)"     "AGE (UCEC)"     </span>
<span class="co">#&gt; [25] "AGE (UCS)"       "AGE (BRCA)"      "AGE (SARC)"      "AGE (TGCT)"     </span>
<span class="co">#&gt; [29] "AGE (THYM)"      "AGE (OV)"        "SMOKING (BLCA)"  "SMOKING (LUAD)" </span>
<span class="co">#&gt; [33] "SMOKING (HNSCC)" "SMOKING (KIRP)"  "SMOKING (PAAD)"  "SMOKING (ESCSQ)"</span>
<span class="co">#&gt; [37] "SMOKING (ESCAD)" "SMOKING (CESC)"  "POLE (UCEC)"     "POLE (STAD)"    </span>
<span class="co">#&gt; [41] "POLE (COAD)"     "POLE (BRCA)"     "MSI (UCEC)"      "MSI (STAD)"     </span>
<span class="co">#&gt; [45] "MSI (COAD)"      "BRCA (BRCA)"     "BRCA (OV)"       "UV* (SKCM)"     </span>
<span class="co">#&gt; [49] "POLD (UCEC)"     "POLD (STAD)"     "MGMT (GBM)"      "MGMT (LGG)"     </span>
<span class="co">#&gt; [53] "IDH (LGG)"       "IDH (GBM)"       "BMI (UCEC)"      "BMI (KIRP)"     </span>
<span class="co">#&gt; [57] "BMI (ESCA)"      "BMI (COAD)"      "ALCOHOL (HNSCC)" "ALCOHOL (ESCA)" </span>
<span class="co">#&gt; [61] "ALCOHOL (LIHC)"  "HepB (LIHC)"     "HepC (LIHC)"     "AAcid (BLCA)"   </span>
<span class="co">#&gt; [65] "Asb* (MESO)"     "APOPEC (CESC)"   "APOPEC (KIRC)"</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use pre-trained signature</span>
<span class="va">newdata</span> <span class="op">=</span> <span class="fu"><a href="../reference/predict_signature.html">predict_signature</a></span><span class="op">(</span><span class="va">supersig_ls</span><span class="op">[[</span><span class="st">"SMOKING (LUAD)"</span><span class="op">]</span><span class="op">]</span>, 
                            newdata <span class="op">=</span> <span class="va">input_dt</span>, factor <span class="op">=</span> <span class="st">"Smoking"</span><span class="op">)</span>
<span class="va">newdata</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">IndVar</span>, <span class="va">X1</span>, <span class="va">X2</span>, <span class="va">X3</span>, <span class="va">score</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 5 x 5</span>
<span class="co">#&gt;   IndVar     X1    X2    X3 score</span>
<span class="co">#&gt;    &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span>
<span class="co">#&gt; 1      1 0.02       0     0 0.305</span>
<span class="co">#&gt; 2      1 0.0182     0     0 0.303</span>
<span class="co">#&gt; 3      1 0          0     0 0.280</span>
<span class="co">#&gt; 4      0 0.0189     0     0 0.304</span>
<span class="co">#&gt; 5      0 0.0417     0     0 0.333</span></code></pre></div>
</div>
<div id="partially-supervised-signatures" class="section level1">
<h1 class="hasAnchor">
<a href="#partially-supervised-signatures" class="anchor"></a>Partially supervised signatures</h1>
<p>In some cases, you may be interested in removing the contribution of a supervised signature from your data frame of mutations as a way to adjust for a particular factor. For example, suppose that we are interested in the deciphering a signature for smoking in lung cancer. We can first remove the contribution of the aging signature in lung cancer, before learning the smoking signature with a supervised or unsupervised method. We discuss in <em>Afsari, et al. (2021, ELife)</em> how doing so can lead to better performance.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">adjusted_dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/partial_signature.html">partial_signature</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">input_dt</span>, object <span class="op">=</span> <span class="va">supersig</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">adjusted_dt</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 5 x 99</span>
<span class="co">#&gt;   IndVar sample_id   AGE `A[T&gt;G]T` `C[T&gt;A]A` `G[C&gt;A]A` `G[C&gt;G]G` `G[C&gt;G]T`</span>
<span class="co">#&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt; 1      1         1    50         1         1    1.04           1    1.04  </span>
<span class="co">#&gt; 2      1         2    55         0         0    0.0411         0    0.0411</span>
<span class="co">#&gt; 3      1         3    72         1         1    0.0538         0    0.0538</span>
<span class="co">#&gt; 4      0         4    53         0         0    0.0396         0    0.0396</span>
<span class="co">#&gt; 5      0         5    48         0         1    1.04           0    0.0359</span>
<span class="co">#&gt; # … with 91 more variables: A[C&gt;G]T &lt;dbl&gt;, C[C&gt;A]T &lt;dbl&gt;, C[T&gt;C]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[C&gt;A]C &lt;dbl&gt;, T[C&gt;A]T &lt;dbl&gt;, A[T&gt;C]C &lt;dbl&gt;, C[T&gt;C]C &lt;dbl&gt;, T[T&gt;A]C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[C&gt;G]G &lt;dbl&gt;, G[C&gt;T]A &lt;dbl&gt;, A[C&gt;A]T &lt;dbl&gt;, C[C&gt;A]C &lt;dbl&gt;, G[T&gt;G]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[C&gt;T]C &lt;dbl&gt;, T[C&gt;T]C &lt;dbl&gt;, A[C&gt;T]C &lt;dbl&gt;, G[C&gt;T]C &lt;dbl&gt;, C[C&gt;T]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[C&gt;T]T &lt;dbl&gt;, A[C&gt;T]T &lt;dbl&gt;, G[C&gt;T]T &lt;dbl&gt;, C[C&gt;T]A &lt;dbl&gt;, T[C&gt;T]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[C&gt;T]A &lt;dbl&gt;, C[C&gt;T]G &lt;dbl&gt;, T[C&gt;T]G &lt;dbl&gt;, A[C&gt;T]G &lt;dbl&gt;, G[C&gt;T]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[C&gt;A]C &lt;dbl&gt;, G[C&gt;A]C &lt;dbl&gt;, G[C&gt;A]T &lt;dbl&gt;, C[C&gt;A]A &lt;dbl&gt;, T[C&gt;A]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[C&gt;A]A &lt;dbl&gt;, C[C&gt;A]G &lt;dbl&gt;, T[C&gt;A]G &lt;dbl&gt;, A[C&gt;A]G &lt;dbl&gt;, G[C&gt;A]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[C&gt;G]C &lt;dbl&gt;, T[C&gt;G]C &lt;dbl&gt;, A[C&gt;G]C &lt;dbl&gt;, G[C&gt;G]C &lt;dbl&gt;, C[C&gt;G]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[C&gt;G]T &lt;dbl&gt;, C[C&gt;G]A &lt;dbl&gt;, T[C&gt;G]A &lt;dbl&gt;, A[C&gt;G]A &lt;dbl&gt;, G[C&gt;G]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[C&gt;G]G &lt;dbl&gt;, A[C&gt;G]G &lt;dbl&gt;, T[T&gt;C]C &lt;dbl&gt;, G[T&gt;C]C &lt;dbl&gt;, C[T&gt;C]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[T&gt;C]T &lt;dbl&gt;, A[T&gt;C]T &lt;dbl&gt;, G[T&gt;C]T &lt;dbl&gt;, C[T&gt;C]A &lt;dbl&gt;, T[T&gt;C]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[T&gt;C]A &lt;dbl&gt;, G[T&gt;C]A &lt;dbl&gt;, T[T&gt;C]G &lt;dbl&gt;, A[T&gt;C]G &lt;dbl&gt;, G[T&gt;C]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[T&gt;A]C &lt;dbl&gt;, A[T&gt;A]C &lt;dbl&gt;, G[T&gt;A]C &lt;dbl&gt;, C[T&gt;A]T &lt;dbl&gt;, T[T&gt;A]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   A[T&gt;A]T &lt;dbl&gt;, G[T&gt;A]T &lt;dbl&gt;, T[T&gt;A]A &lt;dbl&gt;, A[T&gt;A]A &lt;dbl&gt;, G[T&gt;A]A &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[T&gt;A]G &lt;dbl&gt;, T[T&gt;A]G &lt;dbl&gt;, A[T&gt;A]G &lt;dbl&gt;, G[T&gt;A]G &lt;dbl&gt;, C[T&gt;G]C &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[T&gt;G]C &lt;dbl&gt;, A[T&gt;G]C &lt;dbl&gt;, G[T&gt;G]C &lt;dbl&gt;, C[T&gt;G]T &lt;dbl&gt;, T[T&gt;G]T &lt;dbl&gt;,</span>
<span class="co">#&gt; #   C[T&gt;G]A &lt;dbl&gt;, T[T&gt;G]A &lt;dbl&gt;, A[T&gt;G]A &lt;dbl&gt;, G[T&gt;G]A &lt;dbl&gt;, C[T&gt;G]G &lt;dbl&gt;,</span>
<span class="co">#&gt; #   T[T&gt;G]G &lt;dbl&gt;, A[T&gt;G]G &lt;dbl&gt;, G[T&gt;G]G &lt;dbl&gt;</span></code></pre></div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R Under development (unstable) (2021-03-29 r80130)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Mojave 10.14.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    parallel  stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] ggplot2_3.3.3                     dplyr_1.0.5                      </span>
<span class="co">#&gt;  [3] BSgenome.Hsapiens.UCSC.hg19_1.4.3 BSgenome_1.59.2                  </span>
<span class="co">#&gt;  [5] rtracklayer_1.51.5                VariantAnnotation_1.37.1         </span>
<span class="co">#&gt;  [7] Rsamtools_2.7.2                   Biostrings_2.59.2                </span>
<span class="co">#&gt;  [9] XVector_0.31.1                    SummarizedExperiment_1.21.3      </span>
<span class="co">#&gt; [11] Biobase_2.51.0                    GenomicRanges_1.43.4             </span>
<span class="co">#&gt; [13] GenomeInfoDb_1.27.11              IRanges_2.25.7                   </span>
<span class="co">#&gt; [15] S4Vectors_0.29.15                 MatrixGenerics_1.3.1             </span>
<span class="co">#&gt; [17] matrixStats_0.58.0                BiocGenerics_0.37.1              </span>
<span class="co">#&gt; [19] supersigs_0.99.6                 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] colorspace_2.0-0         rjson_0.2.20             ellipsis_0.3.1          </span>
<span class="co">#&gt;   [4] class_7.3-18             rprojroot_2.0.2          fs_1.5.0                </span>
<span class="co">#&gt;   [7] rstudioapi_0.13          farver_2.1.0             listenv_0.8.0           </span>
<span class="co">#&gt;  [10] furrr_0.2.2              bit64_4.0.5              AnnotationDbi_1.53.1    </span>
<span class="co">#&gt;  [13] prodlim_2019.11.13       fansi_0.4.2              lubridate_1.7.10        </span>
<span class="co">#&gt;  [16] codetools_0.2-18         splines_4.1.0            cachem_1.0.4            </span>
<span class="co">#&gt;  [19] knitr_1.31               jsonlite_1.7.2           pROC_1.17.0.1           </span>
<span class="co">#&gt;  [22] caret_6.0-86             dbplyr_2.1.1             png_0.1-7               </span>
<span class="co">#&gt;  [25] compiler_4.1.0           httr_1.4.2               assertthat_0.2.1        </span>
<span class="co">#&gt;  [28] Matrix_1.3-2             fastmap_1.1.0            cli_2.4.0               </span>
<span class="co">#&gt;  [31] prettyunits_1.1.1        htmltools_0.5.1.1        tools_4.1.0             </span>
<span class="co">#&gt;  [34] gtable_0.3.0             glue_1.4.2               GenomeInfoDbData_1.2.4  </span>
<span class="co">#&gt;  [37] reshape2_1.4.4           rappdirs_0.3.3           Rcpp_1.0.6              </span>
<span class="co">#&gt;  [40] jquerylib_0.1.3          pkgdown_1.6.1            vctrs_0.3.7             </span>
<span class="co">#&gt;  [43] nlme_3.1-152             iterators_1.0.13         timeDate_3043.102       </span>
<span class="co">#&gt;  [46] gower_0.2.2              xfun_0.22                stringr_1.4.0           </span>
<span class="co">#&gt;  [49] globals_0.14.0           lifecycle_1.0.0          restfulr_0.0.13         </span>
<span class="co">#&gt;  [52] XML_3.99-0.6             future_1.21.0            zlibbioc_1.37.0         </span>
<span class="co">#&gt;  [55] MASS_7.3-53.1            scales_1.1.1             ipred_0.9-11            </span>
<span class="co">#&gt;  [58] hms_1.0.0                ragg_1.1.2               curl_4.3                </span>
<span class="co">#&gt;  [61] yaml_2.2.1               memoise_2.0.0            sass_0.3.1              </span>
<span class="co">#&gt;  [64] biomaRt_2.47.7           rpart_4.1-15             stringi_1.5.3           </span>
<span class="co">#&gt;  [67] RSQLite_2.2.6            highr_0.8                BiocIO_1.1.2            </span>
<span class="co">#&gt;  [70] desc_1.3.0               foreach_1.5.1            filelock_1.0.2          </span>
<span class="co">#&gt;  [73] GenomicFeatures_1.43.8   BiocParallel_1.25.5      lava_1.6.9              </span>
<span class="co">#&gt;  [76] rlang_0.4.10             pkgconfig_2.0.3          systemfonts_1.0.1       </span>
<span class="co">#&gt;  [79] bitops_1.0-6             rsample_0.0.9            evaluate_0.14           </span>
<span class="co">#&gt;  [82] lattice_0.20-41          purrr_0.3.4              labeling_0.4.2          </span>
<span class="co">#&gt;  [85] GenomicAlignments_1.27.2 recipes_0.1.15           bit_4.0.4               </span>
<span class="co">#&gt;  [88] tidyselect_1.1.0         parallelly_1.24.0        plyr_1.8.6              </span>
<span class="co">#&gt;  [91] magrittr_2.0.1           R6_2.5.0                 generics_0.1.0          </span>
<span class="co">#&gt;  [94] DelayedArray_0.17.10     DBI_1.1.1                pillar_1.6.0            </span>
<span class="co">#&gt;  [97] withr_2.4.1              survival_3.2-10          KEGGREST_1.31.1         </span>
<span class="co">#&gt; [100] RCurl_1.98-1.3           nnet_7.3-15              tibble_3.1.0            </span>
<span class="co">#&gt; [103] crayon_1.4.1             utf8_1.2.1               BiocFileCache_1.99.3    </span>
<span class="co">#&gt; [106] rmarkdown_2.7            progress_1.2.2           grid_4.1.0              </span>
<span class="co">#&gt; [109] data.table_1.14.0        blob_1.2.1               ModelMetrics_1.2.2.2    </span>
<span class="co">#&gt; [112] digest_0.6.27            tidyr_1.1.3              textshaping_0.3.3       </span>
<span class="co">#&gt; [115] munsell_0.5.0            bslib_0.2.4</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://albertkuo.github.io">Albert Kuo</a>, Yifan Zhang, Bahman Afsari, <a href="https://www.cristiantomasetti.com/">Cristian Tomasetti</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
